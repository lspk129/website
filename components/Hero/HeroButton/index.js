import styled from 'styled-components';

const Button = styled.button`
  position: relative;
  display: block;
  margin: 30px auto;
  overflow: hidden;
  border-width: 0;
  outline: none;
  border-radius: 2px;
  box-shadow: 0 1px 4px rgba(0, 0, 0, .6);
  background-color: #ff6666;
  color: #fff;
  transition: background-color .3s;
  &:hover {
    background-color: #fff;
    color: #ff6666
  };
  padding: 15px 50px;
  top: 65%;
  font-size: 1rem;
  font-weight: 600;
  text-transform: uppercase;
`;

export default Button;
