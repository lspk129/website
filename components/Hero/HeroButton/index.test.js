import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import HeroButton from './index';

describe('HeroButton', () => {
  it('renders corretly', () => {
    const wrapper = renderer.create(<HeroButton />);
    const tree = wrapper.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders a button element', () => {
    const wrapper = shallow(<HeroButton />);
    expect(wrapper).toBePresent();
    expect(wrapper).toHaveTagName('button');
  });

  it('renders "get a free consultation" in the content', () => {
    const wrapper = shallow(
      <HeroButton>get a free consultation</HeroButton>,
    );
    expect(wrapper).toBePresent();
    expect(wrapper).toHaveText('get a free consultation');
  });
});
