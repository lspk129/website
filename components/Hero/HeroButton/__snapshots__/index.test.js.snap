// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroButton renders corretly 1`] = `
<button
  className="eLIWcI"
/>
`;
